src/x509/clu_cert_setup.c

```diff
@@ -205,6 +209,255 @@ int wolfCLU_certSetup(int argc, char** argv)
         }
     }
 
+    /* Print out specific parts as requested */
+    if (ret == WOLFCLU_SUCCESS) {
+        if (wolfCLU_checkForArg("-subject", 8, argc, argv) != 0) {
+            printSubject = 1;
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS && printSubject) {
+        char* subject;
+
+        subject = wolfSSL_X509_NAME_oneline(
+                                     wolfSSL_X509_get_subject_name(x509), 0, 0);
+        if (subject != NULL) {
+            wolfSSL_BIO_write(out, subject, (int)XSTRLEN(subject));
+            wolfSSL_BIO_write(out, "\n", (int)XSTRLEN("\n"));
+            XFREE(subject, 0, DYNAMIC_TYPE_OPENSSL);
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS) {
+        if (wolfCLU_checkForArg("-issuer", 7, argc, argv) != 0) {
+            printIssuer = 1;
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS && printIssuer) {
+        char* issuer;
+
+        issuer = wolfSSL_X509_NAME_oneline(
+                                     wolfSSL_X509_get_issuer_name(x509), 0, 0);
+        if (issuer != NULL) {
+            wolfSSL_BIO_write(out, issuer, (int)XSTRLEN(issuer));
+            wolfSSL_BIO_write(out, "\n", (int)XSTRLEN("\n"));
+            XFREE(issuer, 0, DYNAMIC_TYPE_OPENSSL);
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS) {
+        if (wolfCLU_checkForArg("-serial", 7, argc, argv) != 0) {
+            printSerial = 1;
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS && printSerial) {
+        unsigned char serial[32];
+        int  sz = sizeof(serial);
```

Comment:
You should explicitly cast to `int` here. Some compilers will be unhappy with this.